includes/class-wc-gateway-stripe.php

```diff
@@ -1311,4 +1312,25 @@ public function get_setup_help_text() {
 	public function needs_setup() {
 		return ! $this->publishable_key || ! $this->secret_key;
 	}
+
+	/**
+	 * Updates the test mode based on keys provided when setting up the gateway via onboarding.
+	 *
+	 * @return array
+	 */
+	public function update_onboarding_settings( $settings ) {
+		if ( ! empty( $settings['publishable_key'] ) && ! empty( $settings['secret_key'] ) ) {
+			if ( strpos( $settings['publishable_key'], 'test_' ) === 0 || strpos( $settings['secret_key'], 'test_' ) === 0 ) {
```

Comment:
The keys start with `pk_` and  `sk_` respectively, so this check will need to include those (or be modified in some other way).

Reply:
Is this true for the test keys?  I seem to recall the logic in WooCommerce Admin checking for `test_` prefixes if they were test keys.

Reply:
Yup, the test keys are prefixed with `pk_test_` and `sk_test_`

Reply:
Sorry, I see this was `sk_test` but also updated in https://github.com/woocommerce/woocommerce-admin/pull/5353.

I'll update accordingly once we have a decision on the test mode issue below.